src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -48,7 +49,14 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
 
-                if (exeAssembly != null)
+                // in case of SingleFile deployment, Assembly.Location is empty.
+                if (exeAssembly?.Location.Length == 0)
+                {
+                    isSingleFile = true;
```

Comment:
You can probably get rid of this variable entirely and collapse this `if` with the one below.

Reply:
https://github.com/dotnet/runtime/pull/42176#discussion_r488184358

Reply:
It won't work because it needs to go through the `else` case for single file and then we are checking for isSingleFile.